packages/x-date-pickers/src/MonthPicker/MonthPicker.tsx

```diff
@@ -9,7 +9,25 @@ import { useUtils, useNow } from '../internals/hooks/useUtils';
 import { PickerOnChangeFn } from '../internals/hooks/useViews';
 import { MonthPickerClasses, getMonthPickerUtilityClass } from './monthPickerClasses';
 
-export interface MonthPickerProps<TDate> {
+export interface ExportedMonthPickerProps<TDate> {
+  /**
+   * Callback firing on month change @DateIOType.
+   * @template TDate
+   * @param {TDate} month The new year.
+   * @returns {void|Promise} -
+   */
+  onMonthChange?: (month: TDate) => void | Promise<void>;
```

Comment:
Not related to this PR but `onMonthChange` can return a promise and not `onYearChange`, the typing feels wrong.

I will have a look